src/main/java/de/adesso/anki/battle/world/bodies/Vehicle.java

```diff
@@ -240,16 +243,29 @@ public String convertFactsToMessage() {
     		json.put("inv",arr);
 
 
-			//arr = new JSONArray();
-			//for (GameState gameState : this.factsObstacles) {
-			//	arr.put(gameState.getClass().getSimpleName());
-			//}
-    		//json.put("obstacles",arr);
+
+			// 1 ebene tiefer die einzelnen attribute
+			JSONArray allObstacleFacts = new JSONArray();
+			for (GameState gameState : this.factsObstacles) {
+				JSONArray factArr = new JSONArray();
+				if( gameState instanceof ObjectBehind){
+					factArr.put("Behind");
+					factArr.put(((ObjectBehind) gameState).getType()); //obstacle type
+					factArr.put(((ObjectBehind) gameState).getMetersBehind());
+				}
+				if( gameState instanceof ObjectInFront){
+					factArr.put("Front");
+					factArr.put(((ObjectInFront) gameState).getType()); //obstacle type
+					factArr.put(((ObjectInFront) gameState).getMetersInFront());
+				}
+				allObstacleFacts.put(factArr);
+			}
+    		json.put("obstacles",allObstacleFacts);
     		return json.toString();
 
 		} catch (JSONException e) {
-			// TODO Auto-generated catch block
 			e.printStackTrace();

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)